src/libostree/ostree-repo-checkout.c

```diff
@@ -841,19 +841,60 @@ ostree_repo_checkout_tree (OstreeRepo               *self,
  * cache.
  */
 gboolean
-ostree_repo_checkout_tree_at (OstreeRepo                         *self,
+ostree_repo_checkout_tree_at (OstreeRepo                        *self,
                               OstreeRepoCheckoutOptions         *options,
                               int                                destination_dfd,
                               const char                        *destination_path,
                               const char                        *commit,
                               GCancellable                      *cancellable,
                               GError                           **error)
 {
+  OstreeRepoCheckoutAtOptions new_opts = {0, };
+  new_opts.mode = options->mode;
+  new_opts.overwrite_mode = options->overwrite_mode;
+  new_opts.enable_uncompressed_cache = options->enable_uncompressed_cache;
+  new_opts.disable_fsync = options->disable_fsync;
+  new_opts.process_whiteouts = options->process_whiteouts;
+  new_opts.no_copy_fallback = options->no_copy_fallback;
+  new_opts.subpath = options->subpath;
+  new_opts.devino_to_csum_cache = options->devino_to_csum_cache;
+  return ostree_repo_checkout_at (self, &new_opts, destination_dfd,
+                                  destination_path, commit, cancellable, error);
+}
+
+/**
+ * ostree_repo_checkout_at:
+ * @self: Repo
+ * @options: (allow-none): Options
+ * @destination_dfd: Directory FD for destination
+ * @destination_path: Directory for destination
+ * @commit: Checksum for commit
+ * @cancellable: Cancellable
+ * @error: Error
+ *
+ * Similar to ostree_repo_checkout_tree(), but uses directory-relative
```

Comment:
I'd mention here that this also obsoletes `ostree_repo_checkout_tree_at()` due to introspection issues.